src/Engine/babylon.engine.ts

```diff
@@ -5857,30 +5857,45 @@
             this._bindTextureDirectly(this._gl.TEXTURE_CUBE_MAP, null);
 
             // this.resetTextureCache();
+
             texture.isReady = true;
+
+            texture.onLoadedObservable.notifyObservers(texture);

```

Comment:
should not be in the update function else it will be called multiple times
(Checking isReady perhaps?)

Reply:
hmm, perhaps. maybe i should just leave it the way it was before and not call onLoadedObservable. thoughts?

Reply:
onLoaded is fine. But we probalby just need to call it only if texture.isReady was false